### ISSUE TITLE
HTTPS gives all IP's 0.0.0.0

### ISSUE THREAD

**Issue by Firesphere** [piwik/piwik#8721]
All my non-SSL connections work fine, but https gives 0.0.0.0 as origin IP. Doesn't matter what I try. I've been through the FAQ a dozen times, but no result :(

**Comment by tsteur** [comment]
Can you have a look at https://github.com/piwik/piwik/issues/8030#issuecomment-130219649 and send us the output? That'd be really helpful

**Comment by Firesphere** [comment]
Output:
```
array (
  'TMP' => '/srv/users/serverpilot/tmp/piwik',
  'TEMP' => '/srv/users/serverpilot/tmp/piwik',
  'TMPDIR' => '/srv/users/serverpilot/tmp/piwik',
  'PATH' => '/sbin:/usr/sbin:/bin:/usr/bin',
  'USER' => 'serverpilot',
  'HOME' => '/srv/users/serverpilot',
  'FCGI_ROLE' => 'RESPONDER',
  'SCRIPT_URL' => '/test.php',
  'SCRIPT_URI' => 'http://piwik.silverstripe-europe.org/test.php',
  'HTTP_HOST' => 'piwik.silverstripe-europe.org',
  'HTTP_X_FORWARDED_FOR' => '82.197.211.195, 82.197.211.195',
  'HTTP_CONNECTION' => 'close',
  'HTTP_ACCEPT_ENCODING' => 'gzip',
  'HTTP_CF_IPCOUNTRY' => 'NL',
  'HTTP_CF_RAY' => '2223d60a24112bca-AMS',
  'HTTP_X_FORWARDED_PROTO' => 'https',
  'HTTP_CF_VISITOR' => '{"scheme":"https"}',
  'HTTP_X_COOKIESOK' => 'I explicitly accept all cookies',
  'HTTP_USER_AGENT' => 'Mozilla/5.0 (Windows NT 10.0; WOW64; rv:42.0) Gecko/20100101 Firefox/42.0',
  'HTTP_ACCEPT' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
  'HTTP_ACCEPT_LANGUAGE' => 'en-US,en;q=0.5',
  'HTTP_COOKIE' => '__cfduid=dfcf747e2e045261c34aef06187440a861441396002',
  'HTTP_DNT' => '1',
  'HTTP_CF_CONNECTING_IP' => '82.197.211.195',
  'HTTP_CF_UNBUFFERED_UPLOAD' => '0',
  'LD_LIBRARY_PATH' => '/opt/sp/apache/lib',
  'SERVER_SIGNATURE' => '',
  'SERVER_SOFTWARE' => 'Apache/2.4.16 (Unix) OpenSSL/1.0.1f',
  'SERVER_NAME' => 'piwik.silverstripe-europe.org',
  'SERVER_ADDR' => '127.0.0.1',
  'SERVER_PORT' => '80',
  'REMOTE_ADDR' => '82.197.211.195',
  'DOCUMENT_ROOT' => '/srv/users/serverpilot/apps/piwik/public',
  'REQUEST_SCHEME' => 'http',
  'CONTEXT_PREFIX' => '',
  'CONTEXT_DOCUMENT_ROOT' => '/srv/users/serverpilot/apps/piwik/public',
  'SERVER_ADMIN' => 'webmaster@',
  'SCRIPT_FILENAME' => '/srv/users/serverpilot/apps/piwik/public/test.php',
  'REMOTE_PORT' => '39841',
  'GATEWAY_INTERFACE' => 'CGI/1.1',
  'SERVER_PROTOCOL' => 'HTTP/1.0',
  'REQUEST_METHOD' => 'GET',
  'QUERY_STRING' => '',
  'REQUEST_URI' => '/test.php',
  'SCRIPT_NAME' => '/test.php',
  'PHP_SELF' => '/test.php',
  'REQUEST_TIME_FLOAT' => 1441643823.7082689,
  'REQUEST_TIME' => 1441643823,
  'argv' => 
  array (
  ),
  'argc' => 0,
)
```

config.ini.php proxy_client_headers:
```
[General]
;assume_secure_protocol = 1
proxy_client_headers[] = "HTTP_X_FORWARDED_FOR"
proxy_client_headers[] = "HTTP_CF_CONNECTING_IP"
proxy_client_headers[] = "HTTP_CLIENT_IP"
proxy_client_headers[] = "REMOTE_ADDR";
```

**Comment by ieperlingetje** [comment]
I could be wrong, but is the output above not from the http connection instead of https?

**Comment by tsteur** [comment]
@Firesphere I just wanted to have a look but I'm wondering where exactly you get origin IP `0.0.0.0`? I need to know in order to reproduce. Is it during tracking?

**Comment by tsteur** [comment]
Do you have anything configured for `proxy_ips`?

**Comment by mattab** [comment]
@Firesphere do you maybe have answers to our couple questions above?

**Comment by Firesphere** [comment]
@mattab Sorry! I've been a bit busy (I'm in the process of moving from the Netherlands to New Zealand, takes up a lot of time and thoughts)

The IP is (according to my breakpoint) in the IP class itself actually, where there's the default 0.0.0.0, and the headers are then looped.
At the breakpoint before the foreach, the IP seems to be set and recognized, but in the loop itself, it's empty.

I do not have anything specifically configured for proxy_ips, as far as I know. I've only altered/tested/tried with the proxy_client_headers

**Comment by tsteur** [comment]
OK we should try to reproduce, only had a look at the code so far and that looked good. @mattab do you mind setting a milestone? Not sure what to set

**Comment by mattab** [comment]
done: `2.15.1`

**Comment by mattab** [comment]
Note: our FAQ about this topic is https://piwik.org/faq/troubleshooting/faq_17710/

**Comment by tsteur** [comment]
@Firesphere do you mind adding this line https://github.com/piwik/piwik/pull/9099/files#diff-a1ef1fb37277203311f5eed60b826653R119 to `core/IP.php` and see if it changes something for you?

**Comment by Firesphere** [comment]
Sorry for the late reply, but it seems this solves the no-ip issue indeed. I've been testing a bit, but have not yet encountered any 0.0.0.0 errors anymore.

**Comment by mattab** [comment]
Nice to hear & good find @tsteur  :+1:  

Let's also add a test for this use case

**Comment by tsteur** [comment]
There's a test in the PR

**Comment by mattab** [comment]
Nice! this issue will be fixed in 2.15.1
